src/gui/transferlistwidget.cpp

```diff
@@ -487,7 +487,33 @@ void TransferListWidget::copySelectedNames() const
     qApp->clipboard()->setText(torrentNames.join('\n'));
 }
 
-void TransferListWidget::copySelectedHashes() const
+void TransferListWidget::copySelectedInfohashes(const CopyInfohashPolicy policy) const
+{
+    const auto selectedTorrents = getSelectedTorrents();
+    QStringList infoHashes;
+    infoHashes.reserve(selectedTorrents.size());
+    switch (policy)
+    {
+    case CopyInfohashPolicy::Version1:
+        for (const BitTorrent::Torrent *torrent : selectedTorrents)
+        {
+            if (const auto infoHash = torrent->infoHash().v1(); infoHash.isValid())
```

Comment:
I know this subject is a hot potato but the result of this would be confusing. I would suggest you add `-` for non-exist hashes, otherwise user won't be able to match the hashes to the torrents on screen.

Reply:
It depends on what it will be used for. Maybe in order to insert it into some other application for processing? Then, on the contrary, `-` entries will be confusing.
Do you really use it in such a way that you need to match the copied infohashes to the torrents on the screen?
If, like me, you're just speculating about possible usecases, let's just leave it as it is and wait for users feedback.

Reply:
>let's just leave it as it is and wait for users feedback.

OK.